pkg/agent/handler/accessrequest.go

```diff
@@ -122,6 +159,38 @@ func (h *accessRequestHandler) getManagedApp(ar *mv1.AccessRequest) (*v1.Resourc
 	return h.client.GetResource(url)
 }
 
+func (h *accessRequestHandler) addSubscription(ar *mv1.AccessRequest) {
+	// TODO - Use subscription reference subresource on AccessRequest instead of custom subresource
+	// once controller starts to populate it.
+	ars := ar.GetSubResource("x-marketplace-subscription")
+	if ars != nil {
+		arSubscription := ars.(map[string]interface{})
```

Comment:
Can you please check if the type is ok so we don't hit a panic if the type was accidentally `map[string]string`

Reply:
Updated to use util method to get the property value on subresource